lib/lynis/lynistest.pm

```diff
@@ -0,0 +1,289 @@
+
+    # Parse the "section" contents of input file
+    open($rf, "$file") or die "Can not open file, $!";
+
+    # Seek the position to avoid duplicated section names
+    if ($file =~ /baseline/) {
+        seek($rf, $f_position_b, 0);
+    }
+    else {
+        seek($rf, $f_position_c, 0);
+    }
+
+    while (my $line = <$rf>) {
+        $str_orig = $line;
+        $str      = $str_orig;
+
+        if ($str =~ /$s_tok/) {
+            # Found a section
+            if ($found == 1) {
+                # This line of content belongs to another/following section, then exit
+                last;
+            }
+
+            # Replace "\n, \r, \(, \) " in this section for easier regex match
+            $str =~ s/\n//g;
+            $str =~ s/\r//g;
+            $str =~ s/\(/_/g;
+            $str =~ s/\)/_/g;
+
+            $section =~ s/\(/_/g;
+            $section =~ s/\)/_/g;
+
+            my $str_new = substr($str, 4);
+            if ("$section" eq "$str_new") {
+                # Found the section
+                # This line of content belongs to this section, then save
+                push @section_content, $str_orig;
+                $found = 1;
+
+                # Save the position for next search to avoid duplicated section name
+                if ($file =~ /baseline/) {
+                    $f_position_b = tell($rf);
+                }
+                else {
+                    $f_position_c = tell($rf);
+                }
+            }
+        }
+        else {
+            # Found a new line but not a section
+            if ($found == 1) {
+                # This line of content belongs to this section, then save
+                push @section_content, $str_orig;
+            }
+        }
+    }
+
+    if ($file =~ /baseline/) {
+        record_info("FYI: baseline content", "@section_content");
+    }
+    else {
+        record_info("FYI: current content", "@section_content");
+    }
+
+    close($rf) or die "Can not close file, $!";
+
+    return @section_content;
+}
+
+# Compare the contents between "baseline" and "current"
+# input: $found - "1", this section is found in baseline file; "0", not found
+#        $arrar1 - section content of "baseline"
+#        $arrar2 - section content of "current"
+# output: $result - compare results: e.g., "ok", "softfail", "fail"
+sub compare_lynis_section_content {
+    my ($found, $array1, $array2) = @_;
+    my @section_baseline = @$array1;
+    my @section_current  = @$array2;
+
+    my $s_new;
+    my $ret    = 0;
+    my $result = "ok";
+
+    # If an old section then compare baseline and current
+    if ($found) {
+        # Do not use "if (@section_baseline ~~ @section_current) {" to avoid
+        # CI check Error "Smartmatch is experimental"
+        if ("join('', @section_baseline)" eq "join('', @section_current)") {
+            record_info("Same", "Section contents of \"Current\" and \"Baseline\" are the same, exit and pass");
+            $result = "ok";
```

Comment:
You have define      my $result = "ok" in line 231, so  it will be safe to remove this line, right?

Reply:
> You have define my $result = "ok" in line 231, so it will be safe to remove this line, right?

Good question, Since the initial result can be set to anything (e.g., "fail") so I kept the following `$result = "ok";` just for easier code reviewing/maintaining and make the test result checking rule much more clear for us.

Reply:
Then no question here.